### PR TITLE
Feature/games 991/dash and zec

### PR DESCRIPTION
**Pull request by calvinatlan**
### What does this PR do?  
Add dash and zec functionality to wallet_lib

### Link to Jira

[GAMES-991](https://bitcoindotcom.atlassian.net/browse/GAMES-991)

### Deployment notes
I will create pypi account and deploy it myself once it is merged

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,11 +6,13 @@ This library contains classes to work with hot wallet for different cryptocurren
 ### Supported cryptocurrencies
 1. BCH
 2. BTC
+3. Dash
```
Comment (fabasoad) [inline README.md]:
```suggestion
3. DASH
```

**Review by fabasoad — CHANGES_REQUESTED** [review]
Looks good. But requires some changes after Hemen's PR will be merged.
1. Move to adapter approach.
2. CMDAdapter should work fine but RPCAdapter should be tested in scope of ZEC and DASH wallets.
3. Cover code by tests (threshold is 65.85% for now, so coverage shouldn't be less than that).
4. Fix `wallet_factory_test.test_get_all_wallets` test.

**Comment by bubbajoe** [comment]
Test coverage needs to be removed or lowered.

**Review by fabasoad — CHANGES_REQUESTED** [review]
1. Why did you remove `get_addresses`?
2. Why did you change WalletFactory to static implementation?
3. Why did you change WalletFactory implementation (from map to array of wallets)?
4. I would propose to rename WalletInterface to something else. For example, WalletBase. As it's a base _class_ but not interface.
5. CI is failing. Please be sure to cover your code by tests.

**Comment by bubbajoe** [comment]
1. Account's are only supported by BCH (BTC if started in deprecated mode) and probably will be deprecated in the future just like core, zcash and dash.
2. Because the usage was only static, and it looks cleaner.
3. So the ticker symbol is in the class and we base it from that. So if there is a change. We only need to change it in one place.
4. I chose the interface suffix because the usage will be just like an interface, basically you can init it. With base, You can init.
5. I added like 20 something tests, and I removed test as well.

**Review by fabasoad — CHANGES_REQUESTED** [review]
1. Got it. And we don't need it in casinos, right?
2. Got it. Please return it back to non-static. Static classes are kinda bad practice (it's hard to mock a static class in tests, etc.)
3. What kind of change? Pros of map is that you don't need to iterate through all the list (https://bitcoincom.slack.com/archives/GLUPED0QH/p1574929098065200?thread_ts=1574927239.062700&cid=GLUPED0QH). What is the pros of the implemented approach?
4. So, better to rename it then.
5. Here is the coverage:
```bash
----------- coverage: platform win32, python 3.8.0-final-0 -----------
Name                                         Stmts   Miss  Cover
----------------------------------------------------------------
wallet_lib\__init__.py                           5      0   100%
wallet_lib\__main__.py                           0      0   100%
wallet_lib\adapters\__init__.py                  3      0   100%
wallet_lib\adapters\cmd_adapter.py              27      0   100%
wallet_lib\adapters\rpc_adapter.py              36      1    97%
wallet_lib\adapters\wallet_adapter_base.py       3      1    67%
wallet_lib\bch_wallet.py                        51     31    39%
wallet_lib\btc_wallet.py                        49     29    41%
wallet_lib\dash_wallet.py                       51     31    39%
wallet_lib\wallet_exceptions.py                  7      0   100%
wallet_lib\wallet_factory.py                    14      0   100%
wallet_lib\wallet_interface.py                  11      4    64%
wallet_lib\zec_wallet.py                        49     29    41%
----------------------------------------------------------------
TOTAL                                          306    126    59%

FAIL Required test coverage of 65.85% not reached. Total coverage: 58.82%
```
With the code and test provided coverage is lower than it was. For example, zec_wallet is covered with 41% only, dash_wallet - 39%. Try to cover all the functionality to pass the CI check. If you can increase the coverage threshold it would be even better.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -28,7 +28,7 @@ jobs:
           python -m pip install -r requirements.txt
           python -m pip install -e .
           source .envrc
-          pytest --cov=wallet_lib --cov-fail-under=65.85 ./tests/
+          pytest --cov=wallet_lib --cov-fail-under=58.68 ./tests/
```
Comment (fabasoad) [inline .github/workflows/ci.yml]:
```suggestion
          pytest --cov=wallet_lib --cov-fail-under=65.85 ./tests/
```

**Review by fabasoad — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (calvinatlan) [reply .github/workflows/ci.yml]: @fabasoad I'll take care of adding more coverage as soon as an urgent craps ticket is solved

**Review by bubbajoe — APPROVED** [review]
Change minimum coverage to 90%, because it's been brought up to 99.^%

**Review by fabasoad — APPROVED** [review]

**Pull request merged**